discovery-provider/src/tasks/aggregates/__init__.py

```diff
@@ -1,22 +1,41 @@
+import logging
 from inspect import currentframe
 from time import time
+from typing import Optional
 
 from sqlalchemy import text
+from sqlalchemy.orm.session import Session
+from src.models import Block
+from src.utils.prometheus_metric import PrometheusMetric
 from src.utils.update_indexing_checkpoints import (
     get_last_indexed_checkpoint,
     save_indexed_checkpoint,
 )
 
+logger = logging.getLogger(__name__)
 
-def try_updating_aggregate_table(logger, db, redis, table_name, aggregate_func):
+
+def try_updating_aggregate_table(
```

Comment:
can we rename this to like `init_task_and_acquire_lock`

Reply:
Sure thing! ff8097d0